tests/python_package_test/test_engine.py

```diff
@@ -3224,3 +3224,128 @@ def test_force_split_with_feature_fraction(tmp_path):
     for tree in tree_info:
         tree_structure = tree["tree_structure"]
         assert tree_structure['split_feature'] == 0
+
+
+def test_goss_boosting_and_strategy_equivalent():
+    X, y = load_boston(return_X_y=True)
```

Comment:
Please don't remove the check on `mean_squared_error`. Such checks are an important way to detect if a proposed future change might impact statistical performance (see, for example, this discussion: https://github.com/microsoft/LightGBM/pull/4228#issuecomment-862710227).

Since `mean_squared_error` has a scale relative to the scale of the target, I think it makes sense that you'd get such different values for two different datasets.

I just ran `lgb.train()` using the same parameters as this test, but:

* on latest commit to `master` (https://github.com/microsoft/LightGBM/commit/4aaeb22932b72aaff632b40ba596ca2533071ca2)
* using `boosting_type = "goss"`

<details><summary>sample code (click me)</summary>

```python
import lightgbm as lgb
import numpy as np
from sklearn.datasets import make_regression
from sklearn.metrics import mean_squared_error
from sklearn.model_selection import train_test_split

X, y = make_regression(n_samples=10_000, n_features=10, n_informative=5)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
lgb_train = lgb.Dataset(X_train, y_train)
lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train)
params1 = {
    'boosting_type': 'goss',
    'metric': 'l2',
    'verbose': -1
}
evals_result1 = {}
gbm = lgb.train(params1, lgb_train,
                num_boost_round=10,
                valid_sets=lgb_eval,
                callbacks=[lgb.record_evaluation(evals_result1)])

# compute the MSE if the model just predicted the mean of the target from training data
print(np.mean(np.square(y_test - np.mean(y))))

#  check evaluation results on validation data
print(evals_result1)
```

</details>

I observed that for this dataset + set  of parameters, the mean squared error from just using the mean of the target in the training data to predict on the test data is `26348.54`.

The mean squared error on the test data from the final iteration of LightGBM training is `7320.57`, a noticeable improvement.

So I'd recommend a test like, say, `mean_squared_error(y_test, gbm.predict(X_test)) < 7500`. I chose a slightly higher value than the observed validation loss to avoid this test becoming flaky (i.e. failing due to minor variations in performance caused by randomness in the training process).

Please conduct a similar exercise for all the tests on model performance removed in https://github.com/GuangdaLiu/LightGBM/commit/006de872210d046f4097b365e5507c365c9eb24b, and replace them with similar thresholds.

Reply:
Thanks for your suggestion! I ran these tests for multiple times, the mean squared errors vary randomly and the largest one I got is `9800`. So I choose an even safer threshold `10000`.

Reply:
> I ran these tests for multiple times, the mean squared errors vary randomly

I should have said earlier, sorry...please pass `random_state=42` (or any other value you'd like) in `make_regression()`. That will ensure that the same dataset is generated every time these tests run, which should greatly reduce the variation.

Reply:
Fixed in [this commit](https://github.com/GuangdaLiu/LightGBM/commit/beaaf19f711d88e9d31ac1ada5dc63d062638926).